### ISSUE TITLE
ngx-bootstrap documentation sucks!

### ISSUE THREAD

**Issue by Joebeazelman** [valor-software/ng2-bootstrap#5566]
The documentation for ngx-bootstrap sucks!  Do you expect developers to read your mind or to gleen over your source code just to understand how this mediocre mush works? 

Don't know what I am talking about?   I'll give you an example in case you haven't already figured it out. In the following documentation page for dropdowns, there's no discussion of how to databind a selected item (https://valor-software.com/ngx-bootstrap/#/dropdowns). I looked long and hard and there's nothing, nil, none, nada!  This is the most common use case for such dropdown menus: tell me what item the user selected from the drop down!  I refused to believe anyone would put out a control without this capability.

Ironically, in other places, such as Buttons (https://valor-software.com/ngx-bootstrap/#/buttons) you do mention how to databind using the **ngmodel**.  If your dropdown control doesn't support databinding to a selection then say it write it out big and bold so that we can runaway instead of wasting our time. You can cut and paste this text directly to your documentation page to help us:

# OUR DROPDOWN MENU DOESN'T DO ANYTHING USEFUL, DON'T WASTE YOUR TIME

See it is easy to do!  

The other major issue are the stupid and brain dead examples!  If you did a better job of separating what you're mediocre mushware does and what the other mediocre mushware does (bootstrap), it would save us a lot of time.  Stop wasting our time showing us how to use bootstrap!  You wasted an entire section showing us how to use the menu dividers?  Do you think we're stupid or were you looking for fluff to make your crappy documentation look voluminous? 

It's sad to see the gradual descent into mediocrity caused by free and open source software.  Developers can no longer support themselves writing quality software as they did decades ago because it is impossible to compete with free, no matter how much it sucks.  There's no incentive to innovate either as exhibited by the rise of vestiges such as the command-line and primitive text editors like VIM. We're in a vicious cycle where there's no no reason to make anything better and no reason to pay for anything better. 

The worst part of it all is that the large corporations such as Apple, Microsoft and Google, companies who would lance your ball sack if you dare breach their intellectual property, have greatly enriched themselves through open source software. As a double whammy, open source also ensure these pig corporations will never get any competition.

In other words, you're not beyond criticism in the light of this reality.  You're also not benefiting any "community" developing inferior software!

**Comment by daniloff200** [comment]
I think, you spend more than 20-30 minutes, to write this post, full of hate and disrespect.
Instead of that, you could spend this time for writing/updating this **bad** documentation.

It's an opensource library, but, you behave like you paid 100000$ for it, and we all here have to serve you, isn't it? 

Shame on you, Mr. Noname

**Comment by daniloff200** [comment]
I don't even want to dive into this, and try to find something useful in this offensive speech.
If you don't like this library - don't use it.

EASY, MAN

**Comment by etki** [comment]
While i don't support the language, "was not paid for" and offering to spend some time on improving are bad arguments.

Building OSS doesn't mean that there are no responsibilities for built product or that it should be taken for granted. OSS and proprietary software serve the same purpose - to make end user spend less time doing their job, and every feature that is added in library reduces time spent by end users by something like (time of implementing feature / [2..4]) * number of end users. It's irrelevant whether or not is paid for, it's the choice of creator whether to sell software or make it available under some license, not of end user, and often end user don't even have ways to pay for OSS. And as a creator, one has full rights to ignore such issues, but if reasoning is not something like "sadly we have no time for this" - which is sadly true for a lot of projects - a question why such OSS project is maintained at all arises.

Offering to spend some time is a nice way to perform a knockdown on opponent, but let's face it: spending 30 minutes by maintainer may be equal to days of work for one-time contributor. While maintainer knows all ins and outs, contributor that sees code for the first time needs to dig half the codebase to write a proper article, otherwise documentation will be either incorrect or will stay of the same quality.

OSS comes with responsibilities.

**Comment by jorendorff** [comment]
> While i don't support the language, "was not paid for" and offering to spend some time on improving are bad arguments.

Don't worry -- those words weren't meant as a tightly reasoned essay about entitlement and responsibilities in open source project management, thoroughly reflecting the worldview of the author.

Rather, sometimes what's called for is a marginally more polite way of telling someone to f*ck off.

**Comment by jackbravo** [comment]
Relevant: https://youtu.be/VBwWbFpkltg?t=3600

### Jubilee

> "I hear by declare a jubilee for all imagined debt and obligations you think you owe me or anyone in the open source community as a whole. Let no one call upon you to ever feel obligated to repay this vanquished debt. Contribute to the open source community because it brings meaning to your life, because contributing sparks joy. Don't participate if it doesn't. Either way, you are whole and we are square."

-- DHH (redacted to say open source instead of rails)

**Comment by mg217** [comment]
@daniloff200

> Instead of that, you could spend this time for writing/updating this bad documentation.

Correct me if I am wrong, but I think a lot of people have already spent lot of their time trying to contribute (70 PRs open at this moment, some 2 years old):

![image](https://user-images.githubusercontent.com/58844854/70789660-b9a1ad00-1d93-11ea-82d1-27557b0e80f9.png)

So far their time has gone big deal wasted. Isn't that a disrespect to them?

> If you don't like this library - don't use it.

1. This might not have been OP's decision to start using it
2. This might fall beyond OP's competence area to migrate to another library.
3. Would you feel good knowing that people stop using the library because the documentation fails to provide a value?

Lastly, I don't really believe anyone really disrespects OSS authors/maintainers' effort and time they sacrifice, despite the bitter and sometimes offensive words that can be heard. The consumers try to use the library, which is not their beloved fostered baby, but just another product, and they find it difficult, this often leading to frustration at their desks/projects/teams. Rather than throwing a tantrum at an offensive comment, you could sit back and try to understand the problem beneath it. Maybe in some use cases the library (or its documentation) is not as wonderful as you believe it to be? (which in no way defies the effort you had put in it).

 _EASY, MAN_

**Comment by mg217** [comment]
@Etki

> While i don't support the language, "was not paid for" and offering to spend some time on improving are bad arguments.

❤️ 

> OSS and proprietary software serve the same purpose - to make end user spend less time doing their job

❤️ 

It's irrelevant whether or not is paid for, it's the choice of creator whether to sell software or make it available under some license, not of end user, and often end user don't even have ways to pay for OSS.

❤️ 

> Offering to spend some time is a nice way to perform a knockdown on opponent, but let's face it: spending 30 minutes by maintainer may be equal to days of work for one-time contributor. 

❤️ 

> OSS comes with responsibilities.

❤️

**Comment by JimBobSquarePants** [comment]
>Building OSS doesn't mean that there are no responsibilities for built product or that it should be taken for granted.

@etki you couldn't be more incorrect here. From the license. (MIT)

https://github.com/valor-software/ngx-bootstrap/blob/6e200b5f88ff38a3240a7d4550f38825fb198307/LICENSE#L16-L22
